### PR TITLE
[feature] Add gtfs2netexfr binary

### PR DESCRIPTION
**Pull request by woshilapin**

### CODE REVIEW

File: src/bin/gtfs2netexfr.rs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+    /// output directory
+    #[structopt(short = "o", long = "output", parse(from_os_str))]
+    output: PathBuf,
+
+    /// config file
+    #[structopt(short = "c", long = "config", parse(from_os_str))]
+    config_path: Option<PathBuf>,
+
+    /// OnDemandTransport GTFS source
```
Comment (ArnaudOggy) [inline src/bin/gtfs2netexfr.rs]: This on is just for GTFS. You can remote it.

File: src/bin/gtfs2netexfr.rs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    #[structopt(short = "t", long = "on-demand-transport")]
```
Comment (ArnaudOggy) [inline src/bin/gtfs2netexfr.rs]: This on is just for GTFS. You can remote it.

File: src/bin/gtfs2netexfr.rs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    odt: bool,
```
Comment (ArnaudOggy) [inline src/bin/gtfs2netexfr.rs]: This on is just for GTFS. You can remote it.

File: src/bin/gtfs2netexfr.rs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+use transit_model::Result;
+
+#[derive(Debug, StructOpt)]
+#[structopt(name = "gtfs2netexfr", about = "Convert a GTFS to NeTEx France.")]
+struct Opt {
+    /// input directory.
+    #[structopt(short = "i", long = "input", parse(from_os_str), default_value = ".")]
+    input: PathBuf,
+
+    /// output directory
```
Comment (datanel) [inline src/bin/gtfs2netexfr.rs]: or zip file?

File: src/bin/gtfs2netexfr.rs [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+use structopt;
+use structopt::StructOpt;
+use transit_model;
+use transit_model::Result;
+
+#[derive(Debug, StructOpt)]
+#[structopt(name = "gtfs2netexfr", about = "Convert a GTFS to NeTEx France.")]
+struct Opt {
+    /// input directory.
+    #[structopt(short = "i", long = "input", parse(from_os_str), default_value = ".")]
```
Comment (datanel) [inline src/bin/gtfs2netexfr.rs]:
```suggestion
    #[structopt(short, long, parse(from_os_str), default_value = ".")]
```

File: src/bin/gtfs2netexfr.rs [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    input: PathBuf,
+
+    /// output directory
+    #[structopt(short = "o", long = "output", parse(from_os_str))]
```
Comment (datanel) [inline src/bin/gtfs2netexfr.rs]:
```suggestion
    #[structopt(short, long, parse(from_os_str))]
```

File: src/bin/gtfs2netexfr.rs [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+    output: PathBuf,
+
+    /// config file
+    #[structopt(short = "c", long = "config", parse(from_os_str))]
```
Comment (datanel) [inline src/bin/gtfs2netexfr.rs]:
```suggestion
    #[structopt(short, long = "config", parse(from_os_str))]
```

File: src/bin/gtfs2netexfr.rs [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+    config_path: Option<PathBuf>,
+
+    /// OnDemandTransport GTFS source
+    #[structopt(short = "t", long = "on-demand-transport")]
+    odt: bool,
+
+    /// name for the participant
+    #[structopt(short = "p", long = "participant")]
```
Comment (datanel) [inline src/bin/gtfs2netexfr.rs]:
```suggestion
    #[structopt(short, long)]
```

File: src/bin/gtfs2netexfr.rs [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+    participant: String,
+
+    /// code for the provider of stops
+    #[structopt(short = "s", long = "stop-provider")]
```
Comment (datanel) [inline src/bin/gtfs2netexfr.rs]:
```suggestion
    #[structopt(short, long)]
```

File: src/bin/gtfs2netexfr.rs (hunk 3 above)
Comment (woshilapin) [reply src/bin/gtfs2netexfr.rs]: Mmm, I'm not sure, we need a GTFS -> NeTEx France converter, not a NTFS -> NeTEx France one. So I guess we're gonna need this. The question is, can we have a default value, do we want to expose that? @prhod any idea?

File: src/bin/gtfs2netexfr.rs (hunk 4 above)
Comment (woshilapin) [reply src/bin/gtfs2netexfr.rs]: How about doing the same than the other converters for now? If we need, we'll do into a zip file as an additional PR.

File: src/bin/gtfs2netexfr.rs (hunk 3 above)
Comment (prhod) [reply src/bin/gtfs2netexfr.rs]: following a discussion  with @woshilapin, this switch is needed when reading from a GTFS and this is the case here. This flag should be set to "false" if not provided (same default as gtfs2ntfs)

**Comment by woshilapin** [comment]
@datanel I applied all the proposed modifications you made. Thanks.

File: src/bin/gtfs2netexfr.rs [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+        short = "x",
+        long,
+        parse(try_from_str),
+        default_value = &transit_model::CURRENT_DATETIME
+    )]
+    current_datetime: NaiveDateTime,
+}
+
+fn run(opt: Opt) -> Result<()> {
+    info!("Launching gtfs2ntfs...");
```
Comment (ArnaudOggy) [inline src/bin/gtfs2netexfr.rs]: gtfs2netexfr

File: src/bin/gtfs2netexfr.rs (hunk 3 above)
Comment (ArnaudOggy) [reply src/bin/gtfs2netexfr.rs]: Ok, i finally understood :smile:

File: src/bin/gtfs2netexfr.rs (hunk 10 above)
Comment (woshilapin) [reply src/bin/gtfs2netexfr.rs]: Thanks. Done in a46029387de059f6a74b115f99b184da1ca127ac.

File: src/bin/gtfs2netexfr.rs [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+    #[structopt(short = "t", long = "on-demand-transport")]
+    odt: bool,
+
+    /// name for the participant
+    #[structopt(short, long)]
+    participant: String,
+
+    /// code for the provider of stops
+    #[structopt(short, long)]
+    stop_provider: Option<String>,
```
Comment (ArnaudOggy) [inline src/bin/gtfs2netexfr.rs]: It may be relevant to remove the Option and set "LOC" by default

Comment (woshilapin) [reply src/bin/gtfs2netexfr.rs]: Having `LOC` as a default is conversion logic and should not be in `tartare-tools` in my opinion. But yet, there is something interesting in what you said so I made some modifications in `transit_model` (see 3e3602c1a8ee7fda67ab217fd576850ffb7ce91d).

Comment (ArnaudOggy) [reply src/bin/gtfs2netexfr.rs]: Ok :+1:

File: src/bin/gtfs2netexfr.rs [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+    current_datetime: NaiveDateTime,
+}
+
+fn run(opt: Opt) -> Result<()> {
+    info!("Launching gtfs2netexfr...");
+
+    let model = if opt.input.is_file() {
+        transit_model::gtfs::read_from_zip(opt.input, opt.config, None, opt.odt)?
+    } else if opt.input.is_dir() {
+        transit_model::gtfs::read_from_path(opt.input, opt.config_path, None, opt.odt)?
```
Comment (ArnaudOggy) [inline src/bin/gtfs2netexfr.rs]: opt.config and not opt.config_path

Comment (woshilapin) [reply src/bin/gtfs2netexfr.rs]: Indeed, thanks. Fixed.

**Pull request merged**